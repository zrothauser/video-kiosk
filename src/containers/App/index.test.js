// Dependencies
import React from 'react';
import { shallow } from 'enzyme';

// Component to test
import { App } from './index';

describe('containers', () => {
  describe('App', () => {
    it('renders without crashing', () => {
      const mockProps = {
        fetchAppSettings: jest.fn(),
        fetchAppData: jest.fn(),
        isVideoIndexOpen: true,
        toggleVideoIndex: jest.fn(),
        closeVideoIndex: jest.fn(),
        showHeader: true,
        setSlug: 'media-channel',
        defaultSet: 'media-channel',
        homeURL: 'http://video.clyffordstillmuseum.org',
      };

      shallow(<App {...mockProps} />);
    });
  });
});
