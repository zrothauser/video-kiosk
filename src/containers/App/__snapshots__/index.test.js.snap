// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers App renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    closeVideoIndex={[MockFunction]}
    defaultSet="media-channel"
    fetchAppData={
      [MockFunction] {
        "calls": Array [
          Array [
            "media-channel",
          ],
          Array [
            "media-channel",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    fetchAppSettings={[MockFunction]}
    homeURL="http://video.clyffordstillmuseum.org"
    isVideoIndexOpen={true}
    setSlug="media-channel"
    showHeader={true}
    toggleVideoIndex={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header
          closeVideoIndex={[MockFunction]}
          homeURL="http://video.clyffordstillmuseum.org"
          toggleVideoIndex={[MockFunction]}
          visible={true}
        />,
        <main
          className="b-main"
        >
          <Route
            render={[Function]}
          />
        </main>,
        <Connect(VideoIndex)
          visible={true}
        />,
      ],
      "className": "b-app",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "closeVideoIndex": [MockFunction],
          "homeURL": "http://video.clyffordstillmuseum.org",
          "toggleVideoIndex": [MockFunction],
          "visible": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Route
            render={[Function]}
          />,
          "className": "b-main",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "main",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "visible": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header
            closeVideoIndex={[MockFunction]}
            homeURL="http://video.clyffordstillmuseum.org"
            toggleVideoIndex={[MockFunction]}
            visible={true}
          />,
          <main
            className="b-main"
          >
            <Route
              render={[Function]}
            />
          </main>,
          <Connect(VideoIndex)
            visible={true}
          />,
        ],
        "className": "b-app",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "closeVideoIndex": [MockFunction],
            "homeURL": "http://video.clyffordstillmuseum.org",
            "toggleVideoIndex": [MockFunction],
            "visible": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Route
              render={[Function]}
            />,
            "className": "b-main",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "main",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "visible": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
